crates/cli/src/subcommands/generate/rust.rs

```diff
@@ -828,13 +903,14 @@ fn print_reducer_enum_defn(module: &ModuleDef, out: &mut Indenter) {
         "pub enum Reducer {",
         |out| {
             for reducer in iter_reducers(module) {
-                writeln!(
-                    out,
-                    "{}({}::{}),",
-                    reducer_variant_name(&reducer.name),
-                    reducer_module_name(&reducer.name),
-                    reducer_args_type_name(&reducer.name),
-                );
+                write!(out, "{} ", reducer_variant_name(&reducer.name));
+                if !reducer.params_for_generate.elements.is_empty() {
+                    // If the reducer has any arguments, generate a "struct variant."
+                    // If it doesn't, generate a "unit variant" instead.
+                    // For some reason, Rust prohibits empty struct variants.
```

Comment:
Unless I'm misunderstanding something, it doesn't. Enums like

```rust
enum Foo {
    A {}
}
```

are fine.

Reply:
Strange - I tried this and got an error. I wonder if I misunderstood and it was actually from the SATS derive macros, before I removed them.

Reply:
Ah, yep, it was the SATS macros. Well, now that I've done the work to generate unit variants, I figure we might as well keep it - it's better UX. I'll just change the comments.

Reply:
Yeah I agree about better UX either way.